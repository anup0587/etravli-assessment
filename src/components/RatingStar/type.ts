export interface RatingStarProps{
    subTitle?:string,
    averageRating:number
}