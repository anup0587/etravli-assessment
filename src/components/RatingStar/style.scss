.movie_Rating_star{
    display: flex;
    justify-items: center;
    margin-bottom: 1rem;
    gap: 0.5rem;
}