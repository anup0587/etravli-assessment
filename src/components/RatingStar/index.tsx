import { Box, Rating, Typography } from "@mui/material";
import React from "react";
import "./style.scss";
import { RatingStarProps } from "./type";

const RatingStar = ({ subTitle, averageRating }: RatingStarProps) => (
  <Box className="movie_Rating_star" data-testid="ratingStart">
    <Typography variant="subtitle2" gutterBottom>
      {subTitle}
    </Typography>
    <Rating name="rating" defaultValue={averageRating} max={10} />
  </Box>
);

export default RatingStar;
