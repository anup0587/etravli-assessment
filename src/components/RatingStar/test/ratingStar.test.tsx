import React from 'react';
import { render, screen } from '@testing-library/react';
import "@testing-library/jest-dom/extend-expect";
import RatingStar from '..';


describe('RatingStar', () => {
  const props = {
    subTitle: 'Average rating',
    averageRating: 7.5,
  };

  it('matches snapshot', () => {
    const { container } = render(<RatingStar {...props} />);
    expect(container).toMatchSnapshot();
  });
});
