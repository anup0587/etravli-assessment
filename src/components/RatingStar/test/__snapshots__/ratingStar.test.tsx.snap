// Vitest Snapshot v1

exports[`RatingStar > matches snapshot 1`] = `
<div>
  <div
    class="movie_Rating_star MuiBox-root css-0"
    data-testid="ratingStart"
  >
    <h6
      class="MuiTypography-root MuiTypography-subtitle2 MuiTypography-gutterBottom css-1oxen2w-MuiTypography-root"
    >
      Average rating
    </h6>
    <span
      class="MuiRating-root MuiRating-sizeMedium css-iljtu2-MuiRating-root"
    >
      <label
        class="css-dqr9h-MuiRating-label"
        for=":r1:"
      >
        <span
          class="MuiRating-icon MuiRating-iconFilled css-7qmtgc-MuiRating-icon"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
            data-testid="StarIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
            />
          </svg>
        </span>
        <span
          class="MuiRating-visuallyHidden"
        >
          1 Star
        </span>
      </label>
      <input
        class="MuiRating-visuallyHidden"
        id=":r1:"
        name="rating"
        type="radio"
        value="1"
      />
      <label
        class="css-dqr9h-MuiRating-label"
        for=":r2:"
      >
        <span
          class="MuiRating-icon MuiRating-iconFilled css-7qmtgc-MuiRating-icon"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
            data-testid="StarIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
            />
          </svg>
        </span>
        <span
          class="MuiRating-visuallyHidden"
        >
          2 Stars
        </span>
      </label>
      <input
        class="MuiRating-visuallyHidden"
        id=":r2:"
        name="rating"
        type="radio"
        value="2"
      />
      <label
        class="css-dqr9h-MuiRating-label"
        for=":r3:"
      >
        <span
          class="MuiRating-icon MuiRating-iconFilled css-7qmtgc-MuiRating-icon"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
            data-testid="StarIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
            />
          </svg>
        </span>
        <span
          class="MuiRating-visuallyHidden"
        >
          3 Stars
        </span>
      </label>
      <input
        class="MuiRating-visuallyHidden"
        id=":r3:"
        name="rating"
        type="radio"
        value="3"
      />
      <label
        class="css-dqr9h-MuiRating-label"
        for=":r4:"
      >
        <span
          class="MuiRating-icon MuiRating-iconFilled css-7qmtgc-MuiRating-icon"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
            data-testid="StarIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
            />
          </svg>
        </span>
        <span
          class="MuiRating-visuallyHidden"
        >
          4 Stars
        </span>
      </label>
      <input
        class="MuiRating-visuallyHidden"
        id=":r4:"
        name="rating"
        type="radio"
        value="4"
      />
      <label
        class="css-dqr9h-MuiRating-label"
        for=":r5:"
      >
        <span
          class="MuiRating-icon MuiRating-iconFilled css-7qmtgc-MuiRating-icon"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
            data-testid="StarIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
            />
          </svg>
        </span>
        <span
          class="MuiRating-visuallyHidden"
        >
          5 Stars
        </span>
      </label>
      <input
        class="MuiRating-visuallyHidden"
        id=":r5:"
        name="rating"
        type="radio"
        value="5"
      />
      <label
        class="css-dqr9h-MuiRating-label"
        for=":r6:"
      >
        <span
          class="MuiRating-icon MuiRating-iconFilled css-7qmtgc-MuiRating-icon"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
            data-testid="StarIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
            />
          </svg>
        </span>
        <span
          class="MuiRating-visuallyHidden"
        >
          6 Stars
        </span>
      </label>
      <input
        class="MuiRating-visuallyHidden"
        id=":r6:"
        name="rating"
        type="radio"
        value="6"
      />
      <label
        class="css-dqr9h-MuiRating-label"
        for=":r7:"
      >
        <span
          class="MuiRating-icon MuiRating-iconFilled css-7qmtgc-MuiRating-icon"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
            data-testid="StarIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
            />
          </svg>
        </span>
        <span
          class="MuiRating-visuallyHidden"
        >
          7 Stars
        </span>
      </label>
      <input
        class="MuiRating-visuallyHidden"
        id=":r7:"
        name="rating"
        type="radio"
        value="7"
      />
      <label
        class="css-dqr9h-MuiRating-label"
        for=":r8:"
      >
        <span
          class="MuiRating-icon MuiRating-iconFilled css-7qmtgc-MuiRating-icon"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
            data-testid="StarIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
            />
          </svg>
        </span>
        <span
          class="MuiRating-visuallyHidden"
        >
          8 Stars
        </span>
      </label>
      <input
        checked=""
        class="MuiRating-visuallyHidden"
        id=":r8:"
        name="rating"
        type="radio"
        value="8"
      />
      <label
        class="css-dqr9h-MuiRating-label"
        for=":r9:"
      >
        <span
          class="MuiRating-icon MuiRating-iconEmpty css-1c99szj-MuiRating-icon"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
            data-testid="StarBorderIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
            />
          </svg>
        </span>
        <span
          class="MuiRating-visuallyHidden"
        >
          9 Stars
        </span>
      </label>
      <input
        class="MuiRating-visuallyHidden"
        id=":r9:"
        name="rating"
        type="radio"
        value="9"
      />
      <label
        class="css-dqr9h-MuiRating-label"
        for=":ra:"
      >
        <span
          class="MuiRating-icon MuiRating-iconEmpty css-1c99szj-MuiRating-icon"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
            data-testid="StarBorderIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
            />
          </svg>
        </span>
        <span
          class="MuiRating-visuallyHidden"
        >
          10 Stars
        </span>
      </label>
      <input
        class="MuiRating-visuallyHidden"
        id=":ra:"
        name="rating"
        type="radio"
        value="10"
      />
      <label
        class="MuiRating-label MuiRating-pristine css-dqr9h-MuiRating-label"
      >
        <input
          class="MuiRating-visuallyHidden"
          id="rating-empty"
          name="rating"
          type="radio"
          value=""
        />
        <span
          class="MuiRating-visuallyHidden"
        >
          Empty
        </span>
      </label>
    </span>
  </div>
</div>
`;
